sequencing-server/src/app.ts

```diff
@@ -184,58 +183,76 @@ app.get('/health', (_: Request, res: Response) => {
-    parsedDictionaries = pluginManager.getPlugin(getEnv().DICTIONARY_PARSER_PLUGIN).parseDictionary(dictionary)
+  if (
+    pluginManager.hasPlugin(getEnv().DICTIONARY_PARSER_PLUGIN) &&
+    pluginManager.getPlugin(getEnv().DICTIONARY_PARSER_PLUGIN).parseDictionary
+  ) {
+    parsedDictionaries = pluginManager.getPlugin(getEnv().DICTIONARY_PARSER_PLUGIN).parseDictionary(dictionary);
   } else {
-    throw new Error(`POST /dictionary: Plugin - ${getEnv().DICTIONARY_PARSER_PLUGIN} \ndoesn't have a 'parseDictionary' method`);
+    throw new Error(
+      `POST /dictionary: Plugin - ${getEnv().DICTIONARY_PARSER_PLUGIN} \ndoesn't have a 'parseDictionary' method`,
+    );
   }
 
-  let json = {}
+  let json = {};
   for (const dictionaryType of Object.keys(DictionaryType)) {
-    let parsedDictionary : CommandDictionary | ParameterDictionary | ChannelDictionary | undefined;
-    let db_table_name = "";
-    let db_value : any[] = []
+    let parsedDictionary: CommandDictionary | ParameterDictionary | ChannelDictionary | undefined;
+    let db_table_name = '';
+    let db_value: any[] = [];
     if (dictionaryType == DictionaryType.COMMAND && parsedDictionaries.commandDictionary) {
-      db_table_name = 'command_dictionary'
-      parsedDictionary = parsedDictionaries.commandDictionary as CommandDictionary
-    }else if (dictionaryType == DictionaryType.CHANNEL && parsedDictionaries.channelDictionary) {
-      db_table_name = 'channel_dictionary'
-      parsedDictionary = parsedDictionaries.channelDictionary as ChannelDictionary
-    }else if (dictionaryType == DictionaryType.PARAMETER && parsedDictionaries.parameterDictionary) {
-      db_table_name = 'parameter_dictionary'
-      parsedDictionary = parsedDictionaries.parameterDictionary as ParameterDictionary
+      db_table_name = 'command_dictionary';
+      parsedDictionary = parsedDictionaries.commandDictionary as CommandDictionary;
+    } else if (dictionaryType == DictionaryType.CHANNEL && parsedDictionaries.channelDictionary) {
+      db_table_name = 'channel_dictionary';
+      parsedDictionary = parsedDictionaries.channelDictionary as ChannelDictionary;
+    } else if (dictionaryType == DictionaryType.PARAMETER && parsedDictionaries.parameterDictionary) {
+      db_table_name = 'parameter_dictionary';
+      parsedDictionary = parsedDictionaries.parameterDictionary as ParameterDictionary;
     }
-    if (!parsedDictionary){
+    if (!parsedDictionary) {
       continue;
     }
 
     const dictionaryPath = await processDictionary(parsedDictionary, dictionaryType as DictionaryType);
+    let commandDictionaryFilePath = undefined;
+
+    if (persistDictionaryToFilesystem) {
+      // TODO: We'll eventually need to be able to handle dynamic filetypes, right now those aren't passed through in the Hasura action.
+      commandDictionaryFilePath = await writeFile(
+        `command.${parsedDictionary.header.version}.xml`,
```

Comment:
Two questions here:
* this is inside a loop which loops over all three dictionary types, should this `if` have a `&& dictionaryType == DictionaryType.COMMAND` if it's meant to be the command dictionary?
* It makes me a bit uncomfortable to be relying directly on the values from the dictionary headers to generate the path, does it make sense to have fallbacks/checks here or does the dictionary parser give us strong enough guarantees?

Reply:
Per discussion with Cody - I'm OK with assuming these will have values (as guaranteed by another part of the system) but we should make sure they're sanitized to be used as filenames (ie. don't have slashes, etc)

Reply:
Update: discussed with Cody and we decided to make the backend allow uploading *any* kinds of dictionaries, not just command dictionaries, so this code needs to be updated again to allow any type